recipes/libtrace/scripts/collect_generated_problems.py

```
@@ -0,0 +1,224 @@
+        action="store_true",
+        help="Skip invalid lines instead of failing.",
+    )
+    return parser.parse_args()
+
+
+if __name__ == "__main__":
+    args = parse_args()
+    input_files = resolve_input_files(args.input_dir, args.input_files, args.pattern)
+
+    if args.max_problem_tokens is not None and args.max_problem_tokens <= 0:
+        raise ValueError("--max_problem_tokens must be a positive integer or omitted.")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)